### PR TITLE
Prevent duplicate checkbox aria-describedby

### PR DESCRIPTION
**Pull request by colinrotherham**
I've spent a bit of time in the checkbox component recently (passing `describedBy` around) so I've fixed https://github.com/alphagov/govuk-frontend/issues/1248 as it'll benefit our work as well.

### CODE REVIEW

File: src/components/checkboxes/template.njk [hunk 1]
```diff
@@ -55,14 +55,14 @@
     {% set conditionalId = "conditional-" + id %}
     {% set hasHint = true if item.hint.text or item.hint.html %}
     {% set itemHintId = id + '-item-hint' %}
+    {% set itemDescribedBy = describedBy if not hasFieldset else "" %}
+    {% set itemDescribedBy = (describedBy + " " + itemHintId) | trim if hasHint else itemDescribedBy %}
```
Comment (hannalaakso) [inline src/components/checkboxes/template.njk]: Do you think we should consider breaking this out a bit to might make it more readable. Something like:
```
{% set itemDescribedBy = "" %}
{% if not hasFieldset %}
   {% set itemDescribedBy = describedBy %}
{% endif %}
{% if hasHint %}
   {% set itemDescribedBy = (describedBy + " " + itemHintId) | trim %}
{% endif %}
```

Comment (aliuk2012) [reply src/components/checkboxes/template.njk]: I agree with @hannalaakso this was a bit difficult to read. 

Could this not do the same thing you are trying to achieve?

```js
    {% set itemHintId = id + '-item-hint' if hasHint %}
    {% set itemDescribedBy = (describedBy + " " + itemHintId) | trim %}
```

You shouldn't need line 58 if the `itemHintId` is only set if there is a fieldset. If it doesn't have a hint then the last lines `trim` should remove any trailing spaces.

**Comment by colinrotherham** [comment]
Thanks for the feedback. I've been in Nunjucks land a while now, will make it simpler!

File: src/components/checkboxes/template.njk (hunk 1 above)
Comment (colinrotherham) [reply src/components/checkboxes/template.njk]: Sorry @hannalaakso @aliuk2012 both of your versions would affect fieldsets of checkboxes which have an error message, giving us two IDs in `aria-describedby`.

**Original version**
```html
waste-1-item-hint
```

**Our suggestions**
```html
waste-error waste-1-item-hint
```

i.e. The fieldset hint would be applied to both the fieldset and the input, which isn't intended!

Admittedly there is a gap in your test coverage here, so I've added a new example.

Leave it with me.

Comment (colinrotherham) [reply src/components/checkboxes/template.njk]: Is this alright?

```njk
{% set itemHintId = id + "-item-hint" if hasHint else "" %}
{% set itemDescribedBy = describedBy if not hasFieldset else "" %}
{% set itemDescribedBy = (itemDescribedBy + " " + itemHintId) | trim %}
```

All tests pass, including a new one here https://github.com/alphagov/govuk-frontend/pull/1265/commits/d9c4957179324727b261add2345d1324f680f320 that tests for the _checkboxes with error message and hints on items_ edge case.

Comment (colinrotherham) [reply src/components/checkboxes/template.njk]: I've kept `else ""` so we don't end up with `undefined` as a string in `aria-describedby`

**Comment by colinrotherham** [comment]
@aliuk2012 I needed to support this new hybrid case (previously untested) so I've pushed a slightly revised version up: https://govuk-frontend-review-pr-1265.herokuapp.com/components/checkboxes/with-error-message-and-hints-on-items/preview

As mentioned on the diffs, it's to cater for:

1. Shows error message, so `aria-describedby` on fieldset
2. Hint on checkboxes, so `aria-describedby` on inputs
3. Error message ID should not be in input `aria-describedby`

Example

@hannalaakso If you'd still prefer those Nunjucks if/else simplifying further let me know.

**Comment by colinrotherham** [comment]
Rebased with `master`
